Src/IronPython/Runtime/ByteArray.cs

```diff
@@ -269,13 +269,19 @@ public ByteArray center(int width, [BytesLike, NotNull]IList<byte> fillchar)
 
         public string decode(CodeContext context, [NotNull]string encoding = "utf-8", [NotNull]string errors = "strict") {
             lock (this) {
-                return StringOps.RawDecode(context, _bytes, encoding, errors);
+                var mv = new MemoryView(this);
+                // TODO: Move the rest outside lock when wrapping bytearray in memoryview turns byttearray readonly
```

Comment:
Maybe we should make `MemoryView` implement `IDisposable`?

Wrapping  `bytearray` in a `memoryview` does not make it readonly, it should only prevent resizing of the buffer.

Reply:
I like the `IDisposable` idea, although this may change then the `IBufferProtocol` will be refactored to `IPythonBuffer`, which will likely be `IDisposable`. Unless `MemoryView` implements `IPythonBuffer` as well...

As for the readonly issue, I will do some more experiments tomorrow.

Reply:
Even if it changes `IDisposable` probably makes sense for interop. Anyway, this change can probably be done in a separate PR since it's a side issue.

```Python
import array
x = array.array("b", range(5))
with memoryview(x) as mv:
    mv[0] = 20
    assert x[0] == 20
    x[0] = 30
    assert mv[0] == 30
    try:
        x.append(1) # this throws
        assert False
    except BufferError:
        pass
try:
    mv.readonly # this throws
    assert False
except ValueError:
    pass
```